### PR TITLE
Implement "Show hidden files" for all 3 major platforms

### PR DESCRIPTION
**Pull request by oneeyeman1**
Please review and apply.

### CODE REVIEW

File: src/gtk/filedlg.cpp [hunk 1]
```diff
@@ -151,6 +151,20 @@ static void gtk_filedialog_update_preview_callback(GtkFileChooser *chooser,
 
 } // extern "C"
 
+extern "C"
+{
+static gboolean gtk_frame_map_callback( GtkFileChooser *file_chooser,
+                        GdkEvent * WXUNUSED(event),
+                        wxFileDialog *dlg )
+{
+    if( dlg->GetWindowStyle() & wxFD_SHOW_HIDDEN )
```
Comment (vadz) [inline src/gtk/filedlg.cpp]:
```suggestion
    if ( dlg->HasFlag(wxFD_SHOW_HIDDEN) )
```

File: src/gtk/filedlg.cpp [hunk 2]
```diff
@@ -151,6 +151,20 @@ static void gtk_filedialog_update_preview_callback(GtkFileChooser *chooser,
...
+        gtk_file_chooser_set_show_hidden( file_chooser, TRUE );
+    else
```
Comment (vadz) [inline src/gtk/filedlg.cpp]: We don't need the `else` branch, this is the default anyhow.

File: src/msw/filedlg.cpp [hunk 3]
```diff
@@ -443,6 +443,9 @@ int wxFileDialog::ShowModal()
 
     if ( HasFdFlag(wxFD_FILE_MUST_EXIST) )
         msw_flags |= OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;
+
+    if( HasFlag(wxFD_SHOW_HIDDEN) )
```
Comment (vadz) [inline src/msw/filedlg.cpp]:
```suggestion
    if ( HasFlag(wxFD_SHOW_HIDDEN) )
```

File: src/osx/cocoa/filedlg.mm [hunk 4]
```diff
@@ -340,6 +340,7 @@ - (BOOL)panel:(id)sender shouldShowFilename:(NSString *)filename
             modalForWindow: nativeParent modalDelegate: m_sheetDelegate
             didEndSelector: @selector(sheetDidEnd:returnCode:contextInfo:)
             contextInfo: nil];
+        [sPanel setShowsHiddenFiles: (HasFlag( wxFD_SHOW_HIDDEN ) ? YES : NO)];
```
Comment (vadz) [inline src/osx/cocoa/filedlg.mm]:
```suggestion
        [sPanel setShowsHiddenFiles: HasFlag(wxFD_SHOW_HIDDEN) ? YES : NO];
```

(same thing below)

**Review by vadz — COMMENTED** [review]
Looks good to me, thanks, but could you please make an effort to use consistent style? TIA!

File: src/gtk/filedlg.cpp (hunk 2 above)
Comment (oneeyeman1) [reply src/gtk/filedlg.cpp]: @vadz ,
Not according to my testing.
Building the dialog sample and trying "File operations -> Open file" displays everything without the else.
I have Gentoo Linux with GTK+3.22.30.

Comment (vadz) [reply src/gtk/filedlg.cpp]: Very strange, the docs [explicitly say](https://developer.gnome.org/gtk3/stable/GtkFileChooser.html#GtkFileChooser--show-hidden) that this is off by default.

Anyhow, if we do need to always call it, then it'd make sense to write
```
gtk_file_chooser_set_show_hidden(file_chooser, dlg->HasFlag(wxFD_SHOW_HIDDEN));
```

Comment (oneeyeman1) [reply src/gtk/filedlg.cpp]: @vadz ,
I don't know - maybe it just Gentoo patching GTK.

Anyway I fixed it as suggested. Please apply and close the ticket after the build passes.

TIA!

**Comment by vadz** [comment]
You need to define `OFN_FORCESHOWHIDDEN` in `wx/msw/missing.h` to fix the CI build failure.

**Comment by oneeyeman1** [comment]
> 
> 
> You need to define `OFN_FORCESHOWHIDDEN` in `wx/msw/missing.h` to fix the CI build failure.

Should be fixed now - let's wait for the builds to finish and apply.

Thank you.

**Comment by vadz** [comment]
Thanks, I'm merging this after removing commented out code and removing the apparently completely unnecessary `map-event` callback: calling `gtk_file_chooser_set_show_hidden()` directly works just fine for me with both GTK 2 and 3.

**Comment by oneeyeman1** [comment]
@vadz 
> Thanks, I'm merging this after removing commented out code

Oops, sorry,

> and removing the apparently completely unnecessary `map-event` callback: calling `gtk_file_chooser_set_show_hidden()` directly works just fine for me with both GTK 2 and 3.

It may be a bug in my specific GTK+ version or a problem with distro (Gentoo).

Anyway, thnx for applying.

**Pull request closed without merging**